schema/queue/configuration/QueueConfiguration.go

```diff
@@ -31,25 +34,56 @@ type QueueConfig struct {
 	Filters    []MqRouterFilterConfiguration `json:"filters"`
 }
 
-type MessageRouterConfiguration struct {
-	Queues map[string]QueueConfig `json:"queues"`
-
-	Logger zerolog.Logger
-}
-
 func (mrc *MessageRouterConfiguration) Init(path string) error {
 	content, err := os.ReadFile(path) // Read json file
 	if err != nil {
 		mrc.Logger.Error().Err(err).Msg("Json file reading error for QueueConfig")
 		return err
 	}
-	fail := json.Unmarshal(content, mrc)
+	fail := mrc.UnmarshalConfig(content)
 	if fail != nil {
 		mrc.Logger.Error().Err(err).Msg("Deserialization error for QueueConfig")
 		return err
 	}
 	return nil
 }
+func (mrc *MessageRouterConfiguration) UnmarshalConfig(data []byte) error {
```

Comment:
Why do we need this method? Isn't it enough that we defined our custom deserialization for `FilterSpec`?

Reply:
I thought its implementation was necessary and just implement it in separate method instead of adding it all into init() method. And it contains usage of FilterSpec deserialization.

Reply:
We need to create the `QueueConfig` a more "go" way. If we can use standard functionality it is better to use it.

I would do something like this:
```go
func LoadQueueConfig(path string) (MessageRouterConfiguration, error) {
   var cfg MessageRouterConfiguration
   content, err := os.ReadFile(path)
   if err != nil {
      return cfg, err
   }
   err := json.Unmarshal(content, &cfg)
   return cfg, err
}
```

If this unmarshal does not work we should implement `UnmarshalJSON` methods on objects that break the unmarshaling instead of writing custom unmarshalling

Reply:
seems like I had some kind of misunderstanding of UnmarshalJSON behavior. Now I understand redundancy of that method and I have changed that. Thank you.